Components/PalindomeChecker.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+
+export const PalindomeChecker = () => {
+  const [input, setInput] = useState('');
+  const [isPalindrome, setIsPalindrome] = useState(false);
+
+  function isArrayLengthEven (arrLength: number) {
+    if (arrLength % 2 === 0){
+      return true;
+    }
+      return false;
+  }
+
+  const isItAPalindrome = (): void => {
+      let arrToCheck = input.split('');
+      let numOfChecks;
+
+      const isArrayEven = isArrayLengthEven(arrToCheck.length);
+
+      if (isArrayEven) {
+        numOfChecks = arrToCheck.length/2;
+      } else (
+        // There's got to be a smarter way to do this that reads easier
+        numOfChecks = (((arrToCheck.length - 1) / 2) + 1)
+      )
+
+      for (let i = 0; numOfChecks; i++) {
+        let firstLetter = arrToCheck[i];
+        let lastLetter = arrToCheck[arrToCheck.length - (i + 1)]
+
+        if (firstLetter !== lastLetter) {
+          setIsPalindrome(false);
+          return;
+        }
+
+        setIsPalindrome(true);
+        return;
+      }
+  }
+
+  return (
+    <div>
+      <h2>
+        {isPalindrome ? 'Yes, it is a Palindrome!' : "No, it's not a Palindrome."}
+      </h2>
+      <input
+        type="text"
+        placeholder="Enter a phrase"
+        onChange={(e) => setInput(e.target.value.toLowerCase())}
```

Comment:
nit -  you could have one less state variable here. OnBlur if you pass the e.target.val to your function, you will have access to the inputs value. Less state variables = less renders for your rect components. 

`onBlur={e => isItAPalindrome(e.target.value)}`

I love how you sanitized the input here with toLowerCase :D

Reply:
Hmm... I hadn't thought of that before. 

I do really like having the button from a UX perspective, as it gives the user a clear way to check the input. But, it gets slightly funky in this case as it's really just a dummy button.

I've thought about adding a check to the button to disable it until the input changes, but then we're back at square 1 with the same number of state variables.

Ultimately, I think it comes down to how we want to balance performance and usability. Is the goal of this to get the best outright performance?